### PR TITLE
Azure Security Center: update pricings to 2023/01/01

### PR DESCRIPTION
**Pull request by Ihab-Zhaika**
Updating the models and the tests for pricings to align with version  2023/01/01

### CODE REVIEW

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Generated/Models/Code.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.SecurityCenter.Models
+{
+    /// <summary> The operation status code. </summary>
+    public readonly partial struct Code : IEquatable<Code>
```
Comment (archerzz) [inline sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Generated/Models/Code.cs:14]: The name is too generic. Consider appending RP prefix or renaming it, like `OperationStatusCode`.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Generated/Models/Extension.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.ResourceManager.SecurityCenter.Models
+{
+    /// <summary> A plan's extension properties. </summary>
+    public partial class Extension
```
Comment (archerzz) [inline sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Generated/Models/Extension.cs:15]: The name is too generic. Consider appending RP prefix or renaming it, like `SecurityCenterPlanExtension`.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Generated/Models/IsEnabled.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.SecurityCenter.Models
+{
+    /// <summary> Indicates whether the extension is enabled. </summary>
+    public readonly partial struct IsEnabled : IEquatable<IsEnabled>
```
Comment (archerzz) [inline sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Generated/Models/IsEnabled.cs:14]: The name is too generic. Consider appending RP prefix or renaming it, like `ExtensionEnabled`.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Generated/ArmSecurityCenterModelFactory.cs [hunk 4]
```diff
@@ -97,17 +97,48 @@ public static ComplianceResultData ComplianceResultData(ResourceIdentifier id =
...
         /// <param name="replacedBy"> Optional. List of plans that replace this plan. This property exists only if this plan is deprecated. </param>
+        /// <param name="extensions"> Optional. List of extensions offered under a plan. </param>
         /// <returns> A new <see cref="SecurityCenter.SecurityCenterPricingData"/> instance for mocking. </returns>
-        public static SecurityCenterPricingData SecurityCenterPricingData(ResourceIdentifier id = null, string name = null, ResourceType resourceType = default, SystemData systemData = null, SecurityCenterPricingTier? pricingTier = null, string subPlan = null, TimeSpan? freeTrialRemainingTime = null, bool? isDeprecated = null, IEnumerable<string> replacedBy = null)
+        public static SecurityCenterPricingData SecurityCenterPricingData(ResourceIdentifier id = null, string name = null, ResourceType resourceType = default, SystemData systemData = null, SecurityCenterPricingTier? pricingTier = null, string subPlan = null, TimeSpan? freeTrialRemainingTime = null, DateTimeOffset? enablementOn = null, bool? isDeprecated = null, IEnumerable<string> replacedBy = null, IEnumerable<Extension> extensions = null)
```
Comment (archerzz) [inline sdk/securitycenter/Azure.ResourceManager.SecurityCenter/src/Generated/ArmSecurityCenterModelFactory.cs:108]: There are breaking changes in the interfaces. Add manual codes to provide backward compatibility, see compute for reference: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations
  - For track2 mgmt SDK, check: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Azure.ResourceManager.Compute/src/Customize

**Review by archerzz — DISMISSED** [review]
Generated code is not up to date.
- You may need to rebase on the latest main, 
- run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
- run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
- run 'dotnet build /t:GenerateCode' to update the generated code.
- run 'dotnet build /t:GenerateTests' to update the generated test code.
    
To reproduce this error locally, run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory securitycenter'.

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**